cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/CassandraDialect.java

```diff
@@ -96,8 +112,17 @@ public LockingStrategy getLockingStrategy(Lockable lockable, LockMode lockMode)
 
 	private ResultSet bindAndExecute(Object[] columnValues, RegularStatement statement) {
 
+		PreparedStatement preparedStatement;
+		try {
+			preparedStatement = preparedStatementCache.get( statement.getQueryString() );
+			session.prepare( statement );
+		}
+		catch (ExecutionException e) {
+			log.failToPrepareCQL( statement.getQueryString(), e.getCause() );
```

Comment:
Ah, just seeing you already declare the required log method, only the exception returned by it is ignored. It should be 

    throw log.failToPrepareCQL( statement.getQueryString(), e.getCause() );